packages/admin/admin/src/hooks/useAsyncOptionsProps.ts

```diff
@@ -12,18 +13,25 @@ export function useAsyncOptionsProps<T>(loadOptions: () => Promise<T[]>): AsyncO
     const [open, setOpen] = useState(false);
     const [options, setOptions] = useState<T[]>([]);
     const [loading, setLoading] = useState(false);
+    const [error, setError] = useState<Error | null>(null);
 
     const handleOpen = async () => {
         setOpen(true);
         setLoading(true);
-        const newOptions = await loadOptions();
-        setOptions(newOptions);
-        setLoading(false);
+        try {
+            const newOptions = await loadOptions();
+            setOptions(newOptions);
+        } catch (e) {
+            setError(e);
```

Comment:
don't you have to call `setError(null)` somewhere? (to give the user a possibility to re-try)

Reply:
the UX Goal is to have a retry button in the error option. I tended to do this in a follow up pull request, but could add it here as well.

Reply:
I believe the error should be set to null if the options have been loaded correctly.